spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/AvgPooling.scala

```diff
@@ -77,13 +77,20 @@ class AvgPooling(
 
     paddingTL = Array(pt, pl)
     paddingBR = Array(pb, pr)
-    val outputMD = MklDnn.MemoryDescInit(4, Array(n, c, oh, ow), DataType.F32, Memory.Format.any)
+    val outputMD = MklDnn.MemoryDescInit(4, Array(n, c, oh, ow), inputs(0).dataType,
+      Memory.Format.any)
     val description = MklDnn.PoolingForwardDescInit(
-      PropKind.Forward, algKind,
+      PropKind.ForwardScoring, algKind,
```

Comment:
why do you change it?

Reply:
The `Forwardscoring` will be better performance than `Forward`.